jbmc/src/jbmc/jbmc_parse_options.cpp

```diff
@@ -726,6 +726,10 @@ int jbmc_parse_optionst::get_goto_program(
 
     goto_modelt &goto_model = dynamic_cast<goto_modelt &>(*goto_model_ptr);
 
+    // if we added the ansi-c library models here this should also call
+    // add_malloc_may_fail_variable_initializations(goto_model);
+    // here
```

Comment:
:no_entry_sign: I am not sure this comment is helpful. JBMC uses its own models rather than the ansi-c models. Also the java language uses `new`, not malloc. This is parsed from the class file into `ID_java_new` / `ID_java_new_array`, which is then lowered directly into `ID_allocate`. This bypasses the use of a malloc function entirely.

Reply:
@thomasspriggs there are valid reasons why you might want to add the ansi-c library here; e.g. if there was JNI support and you wanted to link your java models to C models.

Reply:
Ok, that make a little more sense, if you were thinking about JNI support. AFAIK JNI support isn't planned for the near future. I also suspect that it would be tricky to implement given the complexities of lazy loading.

Reply:
Yeah, that’s why I didn’t want to put it here in the first place! But if you wanted to load the C library, you’d have to add this code and the flags as well, which I believe was @martin-cs's concern and why he asked for these comments.

Reply:
@thomasspriggs : Yes, as @hannes-steffenhagen-diffblue observed JNI is one thing.  Another is that there are a lot of implicit assumptions in the modelling and instrumentation that the C library is present.  It is likely that future modelling of more sophisticated things may well make use of / depend on the C library and I would hate to see the intermediate representation diverge because of this.

@hannes-steffenhagen-diffblue : yes, my concern is in divergence of how we handle things and the "well it works if you use X but not if you use Y first, then X".  This way the divergence is clear and documented in the relevant place (ideally it would be in `goto-instrument` and `goto-diff` too).